test/cppsuite/tests/search_near_01.cxx

```diff
@@ -0,0 +1,249 @@
+
+#include "test_harness/test.h"
+#include "test_harness/workload/thread_context.h"
+#include "test_harness/thread_manager.h"
+#include "test_harness/workload/random_generator.h"
+#include "test_harness/util/api_const.h"
+
+using namespace test_harness;
+/*
+ * In this test, we want to verify that search_near with prefix enabled only traverses the portion
+ * of the tree that follows the prefix portion of the search key. The test is composed of a populate
+ * phase followed by a read phase. The populate phase will insert a set of random generated keys
+ * with a prefix of aaa -> zzz. The read phase will continously perform prefix search near calls,
+ * and validate that the number of entries traversed is within bounds of the search key.
+ */
+class search_near_01 : public test_harness::test {
+    uint64_t keys_per_prefix = 0;
+    uint32_t srchkey_len = 0;
+    std::string alphabet{"abcdefghijklmnopqrstuvwxyz"};
+    const uint32_t ALPHABET_SIZE = 26;
+    const uint32_t PREFIX_KEY_LEN = 3;
+
+    int64_t
+    get_stat(test_harness::thread_context *tc, int stat_field)
+    {
+        int64_t valuep;
+        /* Open our statistic cursor. */
+        scoped_cursor cursor = tc->session.open_scoped_cursor(STATISTICS_URI);
+
+        const char *desc, *pvalue;
+        cursor->set_key(cursor.get(), stat_field);
+        testutil_check(cursor->search(cursor.get()));
+        testutil_check(cursor->get_value(cursor.get(), &desc, &pvalue, &valuep));
+        testutil_check(cursor->reset(cursor.get()));
+        return valuep;
+    }
+
+    void
+    populate_worker(thread_context *tc)
+    {
+        logger::log_msg(LOG_INFO, "Populate with thread id: " + std::to_string(tc->id));
+        std::string prefix_key;
+        int cmpp;
+        uint64_t collections_per_thread = tc->collection_count;
+        for (int64_t i = 0; i < collections_per_thread; ++i) {
+            collection &coll = tc->db.get_collection(i);
+            scoped_cursor cursor = tc->session.open_scoped_cursor(coll.name.c_str());
+            for (uint64_t j = 0; j < ALPHABET_SIZE; ++j) {
+                for (uint64_t k = 0; k < ALPHABET_SIZE; ++k) {
+                    for (uint64_t count = 0; count < tc->key_count; ++count) {
+                        tc->transaction.begin();
+                        /*
+                         * Generate prefix key of aaa -> zzz, and append a random generated key
+                         * string based on the key size configuration.
+                         */
+                        prefix_key = {alphabet.at(tc->id), alphabet.at(j), alphabet.at(k)};
+                        prefix_key += random_generator::instance().generate_string(tc->key_size);
+                        if (tc->insert(cursor, coll.id, prefix_key)) {
+                            /* We failed to insert, rollback our transaction and retry. */
+                            tc->transaction.rollback();
+                            --count;
+                            continue;
```

Comment:
Do we want to limit the number of retries here?

Reply:
Good idea! added now.